tests/ui_tests/app/page/service/page.py

```diff
@@ -56,24 +72,145 @@ def __init__(self, driver, base_url, cluster_id: int, service_id: int):
+        page.wait_page_is_opened()
+        return page
+
+    @allure.step("Open Components tab by menu click")
+    def open_components_tab(self):
+        """Open Components tab by menu click"""
+        self.find_and_click(ObjectPageMenuLocators.service_components_tab)
+        page = ServiceComponentPage(self.driver, self.base_url, self.cluster_id, self.service_id)
+        page.wait_page_is_opened()
+        return page
+
+    @allure.step("Open Configuration tab by menu click")
+    def open_config_tab(self):
+        """Open Configuration tab by menu click"""
+        self.find_and_click(ObjectPageMenuLocators.config_tab)
+        page = ServiceConfigPage(self.driver, self.base_url, self.cluster_id, self.service_id)
+        page.wait_page_is_opened()
+        return page
+
+    @allure.step("Open Status tab by menu click")
+    def open_status_tab(self):
+        """Open Status tab by menu click"""
+        self.find_and_click(ObjectPageMenuLocators.status_tab)
+        page = ServiceStatusPage(self.driver, self.base_url, self.cluster_id, self.service_id)
+        page.wait_page_is_opened()
+        return page
+
+    @allure.step("Open Import tab by menu click")
+    def open_import_tab(self):
+        """Open Import tab by menu click"""
+        self.find_and_click(ObjectPageMenuLocators.import_tab)
+        page = ServiceImportPage(self.driver, self.base_url, self.cluster_id, self.service_id)
+        page.wait_page_is_opened()
+        return page
+
+    @allure.step("Assert that all main elements on the page are presented")
+    def check_all_elements(self):
+        """Assert all main elements presence"""
+        self.assert_displayed_elements(self.MAIN_ELEMENTS)
+
 
 class ServiceMainPage(ServicePageMixin):
     """Service page Main menu"""
 
     MENU_SUFFIX = 'main'
+    MAIN_ELEMENTS = [
+        ObjectPageLocators.title,
+        ObjectPageLocators.subtitle,
+        ObjectPageLocators.text,
+    ]
+
+
+class ServiceComponentPage(ServicePageMixin):
+    """Service page component menu"""
+
+    MENU_SUFFIX = 'component'
+    MAIN_ELEMENTS = [
+        ObjectPageLocators.title,
+        ObjectPageLocators.subtitle,
+        CommonTable.header,
+        CommonTable.row,
+    ]
+
+    def click_action_btn_in_row(self, row: WebElement):
+        """Click on Action button from the row"""
+        self.find_child(row, ServiceComponentLocators.ComponentRow.actions).click()
+
+    @allure.step("Run action {action_name} for component")
+    def run_action_in_component_row(self, row: WebElement, action_name: str):
+        """Run Action by Action button from the row"""
+        self.click_action_btn_in_row(row)
+        self.wait_element_visible(self.table.locators.ActionPopup.block)
+        self.find_and_click(self.table.locators.ActionPopup.button(action_name))
+        self.wait_element_visible(ActionDialog.body)
+        self.find_and_click(ActionDialog.run)
```

Comment:
wait for the body of ActionDialog or better make a specific function to accept action run dialog. otherwise we may have a lot of flaky copy-paste

Reply:
`self.wait_element_visible(ActionDialog.body) `- i'm waiting, no?

Reply:
i'm waiting one line up, no?

Reply:
Sorry, I've missed "wait for ActionDialog disappearance"

Reply:
add